network/network_linux.go

```diff
@@ -345,23 +355,32 @@ func applyDnsConfig(extIf *externalInterface, ifName string) error {
 				continue
 			}
 
-			buf := fmt.Sprintf("--set-dns=%s", server)
-			setDnsList = setDnsList + " " + buf
+			setDnsList = setDnsList + " " + server
 		}
 
 		if setDnsList != "" {
-			cmd := fmt.Sprintf("systemd-resolve --interface=%s%s", ifName, setDnsList)
-			_, err = p.ExecuteCommand(cmd)
+			// example command on Ubuntu22: resolvectl dns azure0 168.63.129.16
+			resCtlCmd := fmt.Sprintf("resolvectl dns %s%s", ifName, setDnsList)
+			// example command on Ubuntu18: systemd-resolve --interface=azure0 --set-dns=168.63.129.16
+			sysResCtlCmd := fmt.Sprintf("systemd-resolve --interface=%s%s", ifName, "--set-dns="+strings.TrimSpace(setDnsList))
+
+			_, err = p.ExecuteCommand(sysResCtlCmd)
 			if err != nil {
+				_, err = p.ExecuteCommand(resCtlCmd)
 				return err
 			}
 		}
 
 		if extIf.DNSInfo.Suffix != "" {
-			cmd := fmt.Sprintf("systemd-resolve --interface=%s --set-domain=%s", ifName, extIf.DNSInfo.Suffix)
-			_, err = p.ExecuteCommand(cmd)
-		}
+			// example command on Ubuntu22: resolvectl domain azure0 dlw5dhyl2njevcuzgmfubi2oid.bx.internal.cloudapp.net
+			resCtlCmd := fmt.Sprintf("resolvectl domain %s %s", ifName, extIf.DNSInfo.Suffix)
+			// example command on Ubuntu18: systemd-resolve --interface=azure0 --set-domain=dlw5dhyl2njevcuzgmfubi2oid.bx.internal.cloudapp.net
+			sysResCtlCmd := fmt.Sprintf("systemd-resolve --interface=%s --set-domain=%s", ifName, extIf.DNSInfo.Suffix)
 
+			if _, err = p.ExecuteCommand(sysResCtlCmd); err != nil {
```

Comment:
lets do in same order as above.. first try resolveCtl and systemdCtl.. can we log error returned by resolveCtl cmd?